plugins/plugin-java/che-plugin-java-ext-jdt/che-jdt-ext-machine/src/main/java/org/eclipse/che/jdt/rest/CompilerSetupService.java

```diff
@@ -87,4 +115,31 @@ public void setParameters(@QueryParam("projectpath") String projectPath, @NotNul
         //noinspection unchecked
         return options.getMap();
     }
+
+    private void updateCreateSettingsFile(Map<String, String> changedParameters) throws IOException {
+        File settingsFile = Paths.get(settingsDir, preferencesFileName).toFile();
+        createParentDirs(settingsFile);
+
+        String content = "";
+        if (settingsFile.createNewFile()) {
+            for (String parameter : changedParameters.keySet()) {
+                content += parameter + "=" + changedParameters.get(parameter) + "\n";
+            }
+        } else {
+            content = new String(toByteArray(settingsFile));
+            for (String changedProperty : changedParameters.keySet()) {
+                Optional<String> propertyFromFile = readLines(settingsFile, Charset.defaultCharset())
```

Comment:
It looks like you read content of file changedParameters.size()+1 times. Try to avoid this